library/Zend/Db/Adapter/Pdo/Abstract.php

```diff
@@ -302,7 +302,15 @@ protected function _quote($value)
     protected function _beginTransaction()
     {
         $this->_connect();
-        $this->_connection->beginTransaction();
+        if (!$this->_connection->beginTransaction()) {
+            $message = 'Begin transaction failed';
+            // PDO returns no error message for now, but it could be changed in future
+            $errorInfo = $this->_connection->errorInfo();
+            if (!empty($errorInfo[2])) {
+                $message .= ': ' . $errorInfo[2];
+            }
+            throw new Zend_Db_Adapter_Exception($message);
```

Comment:
co přidat i kódy do exception? a index 0 do message?

Reply:
dořešeno na slacku, nemá to smysl, mohli bychom si tím způsobit víc zmatků než užitku, hlavně je teď obojí null/0